src/render/opengl/SDL_render_gl.c

```diff
@@ -1324,6 +1351,21 @@ GL_RunCommandQueue(SDL_Renderer * renderer, SDL_RenderCommand *cmd, void *vertic
         cmd = cmd->next;
     }
 
+    /* Turn off vertex array state when we're done, in case external code
+       relies on it being off. */
+    if (data->drawstate.vertex_array) {
```

Comment:
This is the part of the change I'm the least sure about. I added this code so this change doesn't add any state changes that hang around for people using SDL_Render+GL to get surprised by, but I don't know if that's a real problem to be worried about?